.github/workflows/pre-release.yaml

```diff
@@ -0,0 +1,64 @@
+# This workflow concerns the preparation of the `changeset` PR and keeping it updated by tracking the changes on `main` branch.
```

Comment:
this is still vague. explain what will actually happen, or how a user is supposed to use this workflow

Reply:
You can link to the diagram in the comment. Although I'd still like to see a better summary of the process in the comment (what the user has to do to use this action, e.g. Mark a PR with changeset with a pre-release label. When it's merged, the release process is initiated, creating a PR